src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/Razor/ExtractToComponentCodeActionProviderTest.cs

```diff
@@ -97,41 +95,19 @@ public async Task Handle_SinglePointSelection_ReturnsNotEmpty()
             <h1>Hello, world!</h1>
 
             Welcome to your new app.
-            """;
-        TestFileMarkupParser.GetSpan(contents, out contents, out var selectionSpan);
-
-        var request = new VSCodeActionParams()
-        {
-            TextDocument = new VSTextDocumentIdentifier { Uri = new Uri(documentPath) },
-            Range = VsLspFactory.DefaultRange,
-            Context = new VSInternalCodeActionContext()
-        };
-
-        var context = CreateRazorCodeActionContext(request, selectionSpan, documentPath, contents, supportsFileCreation: true);
-
-        var provider = new ExtractToComponentCodeActionProvider();
-
-        // Act
-        var commandOrCodeActionContainer = await provider.ProvideAsync(context, default);
-
-        // Assert
-        Assert.NotEmpty(commandOrCodeActionContainer);
-    }
+            """);
 
     [Fact]
-    public async Task Handle_InProperMarkup_ReturnsEmpty()
-    {
-        // Arrange
-        var documentPath = "c:/Test.razor";
-        var contents = """
+    public Task Handle_InProperMarkup_ReturnsEmpty()
+        => TestAsync("""
             @page "/"
             
             <PageTitle>Home</PageTitle>
             
             <div id="parent">
                 <div>
                     <h1>Div a title</h1>
-                    <p>Div [||]a par</p>
+                    <p>Div {|selection:|}a par</p>
```

Comment:
I would have expected this to extract the surrounding `<p>` tag

Reply:
hmm... I don't think I agree with that one. Especially with the start/end being the same. Maybe if there was a highlighted section?

Reply:
Happy to wait for user feedback on this. Most of the time I think of code actions as associated with a single cursor position, not a range, so thats probably influencing my thinking.